webapp/lib/controller.dart

```diff
@@ -413,14 +413,29 @@ void updateSystemEventsCharts(List<model.SystemEventsData> filteredSystemEventsD
       systemEventsProjectsData[project] = filteredSystemEventsDataList.where((d) => d.project == project).toList());
   view.contentView.createSystemEventsCharts(systemEventsProjectsData);
 
+  var now = new DateTime.now();
+  
+  Map<ChartPeriodFilters, DateTime> periodFiltersDateMap = {
+    ChartPeriodFilters.alltime: null,
+    ChartPeriodFilters.days1: new DateTime(now.year, now.month, now.day - 1, 00),
+    ChartPeriodFilters.days8: new DateTime(now.year, now.month, now.day - 8, 00),
+    ChartPeriodFilters.days15: new DateTime(now.year, now.month, now.day - 15, 00),
+    ChartPeriodFilters.month1: new DateTime(now.year, now.month - 1 , now.day, 00)
+  };
```

Comment:
Consider taking this out and turn it into a helper function. We might want to use in the future for the other charts as well (and we probably should!)

Something like
```
DateTime getStartDateTimeForPeriod(ChartPeriodFilters period) {
  switch (period) {
    ...
  }
}
```

Reply:
Made this into a helper function as suggested